sops/decrypter.go

```diff
@@ -0,0 +1,49 @@
+package sops
+
+import (
+	"fmt"
+
+	"github.com/getsops/sops/v3/decrypt"
+	"gopkg.in/yaml.v3"
+)
+
+// SopsMetadata represents the SOPS metadata structure in encrypted files
+type SopsMetadata struct {
+	Kms          any    `yaml:"kms,omitempty"`
+	GcpKms       any    `yaml:"gcp_kms,omitempty"`
+	AzureKv      any    `yaml:"azure_kv,omitempty"`
+	LastModified string `yaml:"lastmodified,omitempty"`
+	Mac          string `yaml:"mac,omitempty"`
+	Version      string `yaml:"version,omitempty"`
+}
+
+// IsEncrypted checks if the provided YAML content contains SOPS metadata
+func IsEncrypted(data []byte) bool {
```

Comment:
You're still doing 2 x unmarshal. I don't think you need `IsEncrypted` at all. `DecryptYAML` is perfectly safe and robust on its own. All that has to happen in total is
* you Unmarshal, and return if an error
* If "sops" is there, decrypt
* Else, just return the original data